### ISSUE TITLE
Copy button obscures code

### ISSUE THREAD

**Issue by EmmaDawsonDev** [AccessibleForAll/AccessibleWebDev#307]
**Describe the bug**
Sometimes the copy button obscures the first line of code if the line is long. Suggested fix: add padding to the top of the code block to give the copy button enough space so it never overlaps the code below it.

**Screenshots**
<img width="236" alt="image" src="https://github.com/AccessibleForAll/AccessibleWebDev/assets/57045550/e690feef-35a3-4284-8716-56a762cdcffe">

**Comment by Neha** [comment]
Can I take this issue?

**Comment by EmmaDawsonDev** [comment]
Sure, thanks Neha!

**Comment by DIWAKARKASHYAP** [comment]
@EmmaDawsonDev  how about ,  i show copy button on hover ?

**Comment by DIWAKARKASHYAP** [comment]
@EmmaDawsonDev but i think if we add padding from top then its not look good in desktop 
![image](https://github.com/AccessibleForAll/AccessibleWebDev/assets/94478026/bc7fdd71-ad2a-4db1-ae9d-e87c4a5bb429)
but its ok in mobile 
![image](https://github.com/AccessibleForAll/AccessibleWebDev/assets/94478026/4555079f-8c54-43d0-bbc1-7bac7ad4241a)

**Comment by EmmaDawsonDev** [comment]
Let me take a look at how other sites solve this problem

**Comment by DIWAKARKASHYAP** [comment]
@EmmaDawsonDev take a look on this
https://mui.com/material-ui/react-button/
https://flowbite.com/docs/components/buttons/
https://daisyui.com/components/button-group/

i think very few developer use this website on mobile , so we don't need to compromise design for desktop

**Comment by EmmaDawsonDev** [comment]
It's not something I'm willing to compromise on. Things need to work for every user, irrespective of the device they use or the way they interact with the site. 

Unfortunately the Mui and Daisy examples don't seem to have a solution for this. The tailwind one is more interesting and could be implemented in the design. I think it would be good to have a few more good examples to compare and then sketch out a design for them.

**Comment by DIWAKARKASHYAP** [comment]
@EmmaDawsonDev 
![image](https://github.com/AccessibleForAll/AccessibleWebDev/assets/94478026/7c74ee65-2d58-4e98-990b-68c4f28c5f28)
what about this type ?

**Comment by Neha** [comment]
> It's not something I'm willing to compromise on. Things need to work for every user, irrespective of the device they use or the way they interact with the site.
> 
> Unfortunately the Mui and Daisy examples don't seem to have a solution for this. The flowbite one is more interesting and could be implemented in the design. I think it would be good to have a few more good examples to compare and then sketch out a design for them.

Hello, @EmmaDawsonDev let me know what design I need to refer. Thanks :)

**Comment by EmmaDawsonDev** [comment]
Sure! I'm going to take a look at it tomorrow then I'll let you know 😊

**Comment by EmmaDawsonDev** [comment]
I've had a look around and most websites tend to handle this poorly, just letting the button overlap the code. However, I did find one that I liked which is on hashnode where they add the same amount of padding to the top and the bottom so that the button doesn't overlap and the code snippet still looks even. Here's an [example hashnode blog](https://davidts.hashnode.dev/server-side-rendering-ssr-with-nextjs-benefits-and-implementation-strategies) I found showing code snippets 

<img width="430" alt="Screenshot 2023-06-10 102202" src="https://github.com/AccessibleForAll/AccessibleWebDev/assets/57045550/f4d901bb-51be-4236-bacb-958f85430253">

So let's add an equal padding to the top and the bottom of the code block.
